packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -219,11 +219,6 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateOptions<T
       }
     }
 
-    // If it is the intial render and inputValue isn't controlled nor has an intial value, set input to match current selected key if any
-    if (isInitialRender.current && (props.inputValue === undefined && props.defaultInputValue === undefined)) {
-      resetInputValue();
-    }
```

Comment:
Changing this caused the tests to be updated above, but I discussed with Daniel and we think it's ok. This way the value is known during the initial render, which means the input value can be populated during ssr.